src/tslint-cli.ts

```diff
@@ -178,8 +179,48 @@ const processFile = (file: string) => {
     }
 };
 
-const files = argv._;
+const GLOB_IGNORE: string[] = ["**/node_modules/**", "**/typings/**"];
 
-for (const file of files) {
-    glob.sync(file).forEach(processFile);
-}
+const fileArgs: string[] = argv._;
+const files: string[] = fileArgs.reduce((list: string[], arg: string) => {
+    return list.concat(glob.sync(arg, {ignore: GLOB_IGNORE}));
+}, []);
+
+files.forEach(arg => {
+    const stat = fs.lstatSync(arg);
+
+    // Check whether argument is actually a directory
+    if (stat.isDirectory()) {
+        let globIgnore = GLOB_IGNORE;
+
+        // Try finding a tsconfig file
```

Comment:
It feels a little strange to add half-baked support for `tsconfig.json` right now -- I'm worried it might be unintuitive. I think I'd rather have that implemented fully with #858 and a `--project` option. And as @blakeembrey mentioned, we should have a separate `--exclude` option for TSLint to override `tsconfig.json`.

I'd be happy to look at another PR that adds an `--exclude` option, but don't think it should be mixed in with this one.